### PR TITLE
Refactor Signature Validation

### PR DESCRIPTION
**Pull request by nlordell**
This PR is a follow-up to #1871

Specifically, that PR added error handling for the `eth_call` simulation such that reverts are properly classified as `SignatureValidationError::Invalid`. The `Signatures.sol` contract was _supposed to_ handle all reverts from signature verifiers, so that error classification shouldn't have been needed.

**However**, it turns out I made some assumptions about the Solidity `try ... catch` semantics that turned out to be invalid and cause the `Signatures.sol` to still revert in some cases. In particular, `Signatures::validate` will revert if:
- `signer` is not a contract
- `signer.isValidSignature` returns less than 32 bytes of data

With this in mind, it doesn't really make sense to try and handle revert errors in both the Solidity code and the Rust code, so just handle all revert errors in Rust in the `signature_validation::simulation::Validator` code and get rid of the `try ... catch` in the Solidity code.

I also refactored the `From<ethcontract::ExecutionError>` implementation into the existing `From<web3::Error>` (which was no longer being used, and IMO made more sense to be there).

Furthermore, https://github.com/cowprotocol/services/pull/1842 removed some `signature_validator::{arguments, web3}` submodules whose files were accidentally left behind, so I removed them here as well.

### Test Plan

CI - the existing E2E tests for signature validation continue to pass, and added a new one to check that errors are correctly detected.

### CODE REVIEW

File: crates/shared/src/signature_validator/arguments.rs [hunk 1]
```diff
@@ -1,66 +0,0 @@
-use {
```
Comment (nlordell) [inline crates/shared/src/signature_validator/arguments.rs:1]: This module had already been removed in #1842, but the code was accidentally left behind.

File: crates/shared/src/signature_validator/web3.rs [hunk 2]
```diff
@@ -1,108 +0,0 @@
-use {
```
Comment (nlordell) [inline crates/shared/src/signature_validator/web3.rs:1]: This module had already been removed in #1842, but the code was accidentally left behind.

File: crates/shared/src/signature_validator/simulation.rs [hunk 3]
```diff
@@ -106,51 +96,39 @@ impl SignatureValidating for Validator {
...
-impl SimulationResult {
-    fn decode(value: u8) -> Result<Self> {
-        match value {
-            0 => Ok(Self::Ok),
-            1 => Ok(Self::Invalid),
-            2 => Ok(Self::Revert),
-            _ => anyhow::bail!("invalid simulation result variant {value}"),
+impl From<web3::Error> for SignatureValidationError {
+    fn from(err: web3::Error) -> Self {
+        // TODO: This is needed to parse Hardhat revert errors, which
```
Comment (nlordell) [inline crates/shared/src/signature_validator/simulation.rs:117]: Fixed in https://github.com/cowprotocol/ethcontract-rs/pull/962

Will remove this extra branch when the new `ethcontract` version hits.

File: crates/contracts/solidity/Signatures.sol [hunk 4]
```diff
@@ -44,16 +39,10 @@ contract Signatures {
...
-                : Result.INVALID;
-        }
-        catch {
-            gasUsed = gasUsed - gasleft();
-            result = Result.REVERT;
-        }
+        bytes4 magicValue = signer.isValidSignature(order, signature);
+        gasUsed = gasUsed - gasleft();
+
+        require(magicValue == ERC1271_MAGICVALUE, "didn't say the magic word");
```
Comment (sunce86) [inline crates/contracts/solidity/Signatures.sol:45]: 😄

File: crates/shared/src/signature_validator/simulation.rs (hunk 3 above)
Comment (sunce86) [reply crates/shared/src/signature_validator/simulation.rs:117]: Probably worth creating issue so we don't forget.

**Review by sunce86 — APPROVED** [review]
Nice.

File: crates/shared/src/signature_validator/simulation.rs (hunk 3 above)
Comment (nlordell) [reply crates/shared/src/signature_validator/simulation.rs:117]: I already made the PR and plan on doing it today. Can create an issue if you think its worth it.

**Review by MartinquaXD — APPROVED** [review]

File: crates/shared/src/signature_validator/simulation.rs (hunk 3 above)
Comment (nlordell) [reply crates/shared/src/signature_validator/simulation.rs:117]: https://github.com/cowprotocol/services/issues/1875

**Pull request merged**